animalcontrol.lua

```diff
@@ -0,0 +1,211 @@
+                           Note: cannot be used in conjunction with other selection options.
+
+    id <value>           - Selects the unit with the specified id value provided.
+    race <value>         - Selects units which match the race value provided.
+
+    markedfor <value>    - Selects units which have been marked for the value provided.
+                           valid values: slaughter, gelding
+
+    notmarkedfor <value> - Selects units which have not been marked for the value provided.
+                           valid values: slaughter, gelding
+
+    gelded               - Selects units which have already been gelded.
+    notgelded            - Selects units which have not been gelded.
+    male                 - Selects units which are male.
+    female               - Selects units which are female.
+
+-~~~~~~~~~~~
+command options:
+
+    showstats            - Displays physical attributes of the selected animals.
+
+    markfor <value>      - Marks selected animals for the value provided.
+                           valid values: slaughter, gelding
+
+    unmarkfor <value>    - Unmarks selected animals for the value provided.
+                           valid values: slaughter, gelding
+
+other options:
+    help                 - Displays this information
+
+]====]
+
+local Units = df.global.world.units.active
+
+if args.race and not tonumber(args.race) then
+    args.race=string.upper(args.race)
+    local raceID
+    for i,c in ipairs(df.global.world.raws.creatures.all) do
+      if c.creature_id == args.race then
+        raceID = i
+        break
+      end
+    end
+    if not raceID then
+      qerror('Invalid race: ' .. args.race)
+    end
+    args.race = raceID
+end
+
+function safe_pairs(item, keys_only)
+    if keys_only then
+        local mt = debug.getmetatable(item)
+        if mt and mt._index_table then
+            local idx = 0
+            return function()
+                idx = idx + 1
+                if mt._index_table[idx] then
+                    return mt._index_table[idx]
+                end
+            end
+        end
+    end
+    local ret = table.pack(pcall(function() return pairs(item) end))
+    local ok = ret[1]
+    table.remove(ret, 1)
+    if ok then
+        return table.unpack(ret)
+    else
+        return function() end
+    end
+end
+bfilters=(args.id or args.race or args.markedfor or args.notmarkedfor or args.gelded or args.notgelded or args.male or args.female)
+bcommands=(args.showstats or args.markfor or args.unmarkfor)
+bvalid=(args.all and not bfilters) or (not args.all and (bfilters or bcommands))
+
+if args.help or not bvalid then
+    print(help)
+else
+    count=0
+    if args.showstats then
+        print(string.format("%-20s %-6s %-9s %-9s %-5s %-22s %-8s %-25s %-7s %-7s %-7s %-7s %-7s %-7s"
+            , "animal type", "id", "unit id", "race id", "sex", "marked for slaughter", "gelded", "marked for gelding"
+            ,"str","agi","tgh","endur","recup","disres"))
```

Comment:
I'm not sure what some of these abbreviations mean - particularly, I'm guessing "tgh" is "toughness", but I haven't been able to figure out "disres" yet. I understand why you want to keep them short, so maybe you could put explanations in the script documentation.

Reply:
Disease resistance